common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -850,18 +845,18 @@ public static Inet6Address getByName(CharSequence ip, boolean ipv4Mapped) {
 
         final boolean isCompressed = compressBegin > 0;
         // Finish up last set of data that was accumulated in the loop (or before the loop)
-        if (ipv4Seperators > 0) {
+        if (ipv4Separators > 0) {
             if (begin > 0 && i - begin > IPV4_MAX_CHAR_BETWEEN_SEPARATOR ||
-                    ipv4Seperators != IPV4_SEPARATORS ||
+                    ipv4Separators != IPV4_SEPARATORS ||
                     currentIndex >= bytes.length) {
                 return null;
             }
-            if (ipv6Seperators == 0) {
+            if (ipv6Separators == 0) {
                 compressLength = 12;
-            } else if (ipv6Seperators >= IPV6_MIN_SEPARATORS &&
+            } else if (ipv6Separators >= IPV6_MIN_SEPARATORS &&
                            ip.charAt(ipLength - 1) != ':' &&
```

Comment:
this check - `ip.charAt(ipLength - 1) != ':'` - seems superfluous or wrong for the `ipv4Separators > 0`.